run torch::deploy test using pip install

PaliC opened the pull request:
This PR runs the unit tests for [multipy](https://github.com/pytorch/multipy) in pytorch core such that we are able to make sure changes in core do not break multipy as adding `_prims` did.

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #86507

PaliC committed:
run torch::deploy test using pip install

[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"

This PR runs the unit tests for [multipy](https://github.com/pytorch/multipy) in pytorch core such that we are able to make sure changes in core do not break multipy as adding `_prims` did.




[ghstack-poisoned]

anirbanr-fb-r2p reviewed (approved):
Lg in general, other than the cmake update steps.

d4l3k commented on .github/workflows/pull.yml:
```diff
@@ -172,6 +172,8 @@ jobs:
           { config: "distributed", shard: 3, num_shards: 3, runner: "linux.8xlarge.nvidia.gpu" },
           { config: "functorch", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
+          { config: "deploy", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
+          { config: "deploy", shard: 1, num_shards: 1, runner: "linux.4xlarge.nvidia.gpu" },
```
do we need GPUs for these tests? or is CPU sufficient?

PaliC replied:
We only need 1 gpu (good catch on that). However, we are planning on adding gpu tests to do this right, for which we'd want the gpu. We talked about this offline.

d4l3k commented on .jenkins/pytorch/common_utils.sh:
```diff
@@ -137,6 +137,44 @@ function install_triton() {
+function setup_torchdeploy_deps(){
+  conda install -y cmake
+  conda install -y -c conda-forge libpython-static=3.10
+  sudo wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | sudo gpg --dearmor -o /usr/share/keyrings/magic-key.gpg && \
+  sudo echo "deb [arch=amd64,arm64 signed-by=/usr/share/keyrings/magic-key.gpg] https://apt.kitware.com/ubuntu/ bionic main" | sudo tee -a /etc/apt/sources.list && \
+  sudo echo "deb http://security.ubuntu.com/ubuntu focal-security main" | sudo tee -a /etc/apt/sources.list && \
+  sudo apt update && \
+  sudo apt install -y binutils && \
```
after https://github.com/pytorch/multipy/pull/216 this shouldn't be required anymore

PaliC replied:
will update

d4l3k commented on .jenkins/pytorch/common_utils.sh:
```diff
@@ -137,6 +137,44 @@ function install_triton() {
+  local CC="$(which gcc)"
+  local CXX="$(which g++)"
+  export CC
+  export CXX
```
why `local` and not just `export CC`?

PaliC replied:
leads to a lint error :(

d4l3k commented on .jenkins/pytorch/common_utils.sh:
```diff
@@ -137,6 +137,44 @@ function install_triton() {
+  local commit
+  setup_torchdeploy_deps
+  pushd ..
+  git clone https://github.com/pytorch/multipy
```
```
git clone --recurse-submodules https://github.com/pytorch/multipy.git
```

d4l3k reviewed (approved):
LGTM from multipy side -- lint is failing? Probably want someone from core to review as well

PaliC committed:
Update on "run torch::deploy test using pip install"

This PR runs the unit tests for [multipy](https://github.com/pytorch/multipy) in pytorch core such that we are able to make sure changes in core do not break multipy as adding `_prims` did.




[ghstack-poisoned]

PaliC committed:
Update on "run torch::deploy test using pip install"

This PR runs the unit tests for [multipy](https://github.com/pytorch/multipy) in pytorch core such that we are able to make sure changes in core do not break multipy as adding `_prims` did.




[ghstack-poisoned]

PaliC commented:
@pytorchbot merge -g
